### DISCUSSION TITLE
Hiding scroll bar in VSCode webview text area while keeping the scroll functionality intact.

### DISCUSSION THREAD

**Discussion by amitjoshi438** [microsoft/vscode-discussions#843] [Extension Development QnA] [answered]
I have created a text area and want to hide scroll bar while keeping the scroll functionality intact. Usually `::-webkit-scrollbar` is used on browsers, is it supported for webviews also?
![image](https://github.com/microsoft/vscode-discussions/assets/54068463/c482ee3b-73c8-463b-9621-9dde126649ed)

**Comment by r3m0t** [comment, marked as answer]
If you go to Help > About you'll see the Chromium version used, you can use this to look up support for individual features.

You can also use "Developer: Open Webview Developer Tools" and see things like `<style>` and `<script>` tags injected by VS Code into your webview HTML which can affect things. Eg you might need to add `!important` to some CSS declarations or use more specific selectors.

**Comment by theReynald** [comment]
Hi @amitjoshi438, 

 In Visual Studio Code webviews, including webview panels created by extensions, the rendering of HTML and CSS is based on the Electron framework. Electron, in turn, uses the Chromium browser engine for rendering web content. This means that many CSS properties and features supported by modern browsers, including Chrome, are also supported in Visual Studio Code webviews.

Regarding scrollbar styling using `::-webkit-scrollbar`, you can apply this style in webview panels in Visual Studio Code, and it should work as expected because it's based on the Chromium engine. Here's an example of how you can hide the scrollbar while keeping the scroll functionality intact:

```html
<style>
    /* Hide the vertical scrollbar */
    ::-webkit-scrollbar {
        width: 0;
    }

    /* Optional: Style the scrollbar track and thumb */
    ::-webkit-scrollbar-track {
        background: #f1f1f1;
    }

    ::-webkit-scrollbar-thumb {
        background: #888;
    }
</style>
```

In this example, `::-webkit-scrollbar` is used to hide the vertical scrollbar, and you can further style the scrollbar track and thumb as needed.

Keep in mind that while this styling technique should work in most cases, there might be specific scenarios or custom styles in your extension or webview content that could affect the styling of the scrollbar. Testing and adjusting the styles in your specific context may be necessary to achieve the desired result.

Hope this helps.
